### DISCUSSION TITLE
Feature Request: Adding hardware ID field to json data where possible

### DISCUSSION THREAD

**Discussion by roboyeti** [LibreHardwareMonitor/LibreHardwareMonitor#620] [Ideas]
Example, it would be fantastic if the GPUs included their Bus ID.  This would allow the data pulled from json to be connected to other information regarding the GPU (for example, a cryptocurrency mining program will provide hash rate for a GPU, but sometimes not include temp/fan information).  I am looking to use LibreHardwareMonitor to allow me to provide extra info for a particular miner module, so even if the miner doesn't have the GPU specifics, I can connect them with data from here using Bus ID.

**Comment by nathanAjacobs** [comment]
I agree, Bus ID would be useful for GPUs. Not sure how it would be implemented into the IHardware interface though.

The main reason I want to access the Bus ID is so I can display GPUs in a similar fashion to how the Windows Task Manager displays them (i.e. GPU 0, GPU 1, GPU 2).

For my case, I downloaded the library and added it to the IHardware interface as a string and default it to "Not Supported". Then in the NvidiaGpu and AmdGpu classes I set it in their constructors, since it already exists in those classes. I won't even attempt a pull request, because I doubt this is how the main developers would even want it implemented or if they want it implemented at all.
